akka-javasdk/src/main/java/akka/javasdk/AsyncUtils.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk;
+
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+public interface AsyncUtils {
```

Comment:
The idea behind this class is to provide more than just a retry helper method.

Reply:
Good idea. But never sure about resorting to `SomethingUtils`. `AsyncHelpers` is still similar. I guess `AsyncOperations` is more descriptive. Or `AsyncTasks`, `AsyncCalls`, `AsyncCaller`, ...

Reply:
Yeah, AsyncOperations or maybe AsyncOps (scala style), let's wait for others' feedback

Reply:
I would avoid the world Async in the name of the class. 

We can use that same class to retry any call using virtual threads. 

Btw, if we do so, we will have name clashes because of erasure.

```java
<T> T retry(Callable<T> call, int maxRetries)

<T> CompletionStage<T> retry(Callable<CompletionStage<T>> call, int maxRetries)
```

Maybe we need to have:

```java
<T> T retry(Callable<T> call, int maxRetries)
<T> CompletionStage<T> retryAsync(Callable<CompletionStage<T>> call, int maxRetries)
```

`Retries` sounds good to me, but then `Retries.sequence` won't work.
Alternative could be `CallOps`

* `CallOps.retry`
* `CallOps.retryAsync`
* `CallOps.sequence`

Reply:
Or `CallableOps` because it's about `Callables`

Reply:
Well, looking at it once more I realise that the `sequence` do not fit here. It's just about CompletionStage and not about retry things.

Reply:
Added retry without futures, but I see one issue, you can call `retries.retry(() -> CompletableFuture.completedFuture(something))` (should be retries.retryAsync) it will compile fine, but it will not work as expected unless I do some additional checking of types.

Reply:
I don't think we should care for this kind of error. 

If someone call it like that, they get a success return, an CompletionStage and there is nothing to retry. 

If you check the types, you can log or throw, but that's runtime thing so it's already too late. 

Alternatively, we could have one single `retry` and we check if the passed type is a CompletionStage or not. I don't know if I like it. Sounds too much magic.

Reply:
what I mean in my last comment with a single `retry` is that we can check the type and branch between async or sync code internally.